FWCore/Modules/src/BunchCrossingFilter.cc

```diff
@@ -0,0 +1,46 @@
+
+#include "FWCore/Framework/interface/global/EDFilter.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+template <typename T>
+std::vector<T> sorted(std::vector<T> v) {
+  std::sort(v.begin(), v.end());
+  return v;
+}
+
+namespace edm {
+  class BunchCrossingFilter : public global::EDFilter<> {
+  public:
+    explicit BunchCrossingFilter(ParameterSet const& config);
+
+    static void fillDescriptions(ConfigurationDescriptions& descriptions);
+    bool filter(StreamID, Event& event, EventSetup const&) const final;
+
+  private:
+    const std::vector<unsigned int> bunches_;
+  };
+
+  BunchCrossingFilter::BunchCrossingFilter(ParameterSet const& config) :
+    bunches_(sorted(config.getParameter<std::vector<unsigned int>>("bunches")))
+  {
+  }
+
+  bool BunchCrossingFilter::filter(StreamID, Event& event, EventSetup const&) const
+  {
+    return std::binary_search(bunches_.begin(), bunches_.end(), event.bunchCrossing());
+  }
+
+  void
+  BunchCrossingFilter::fillDescriptions(ConfigurationDescriptions& descriptions) {
+    ParameterSetDescription desc;
+    desc.add<std::vector<unsigned int>>("bunches", {})->setComment("Accept an event if its bunch crossing is in the list.");
```

Comment:
So you want the default to be reject all buches if non are specified?

Reply:
Yiu must provide a default. I fno default, the generated cfi files will not contain the parameter, and thus it will not be known to ConfDB after parsing!

Reply:
The default is an empty list.